packages/ui/src/forum/components/threads/ThreadItemBreadcrumbs.tsx

```diff
@@ -1,43 +1,129 @@
 import { useForumMultiQueryCategoryBreadCrumbs } from '@/forum/hooks/useForumMultiQueryCategoryBreadCrumbs'
 
+const MIN_BREADCRUMB_WIDTH = 75
+const MAX_BREADCRUMB_WIDTH = 150
+
 interface ThreadItemBreadcrumbsProps {
   id: string
   nonInteractive?: boolean
 }
 export const ThreadItemBreadcrumbs = memo(({ id, nonInteractive }: ThreadItemBreadcrumbsProps) => {
-  const { breadcrumbs } = useForumMultiQueryCategoryBreadCrumbs(id)
+  const { isLoading, breadcrumbs } = useForumMultiQueryCategoryBreadCrumbs(id)
 
-  return (
-    <ThreadItemBreadcrumbsList>
-      {nonInteractive ? (
-        <BreadcrumbsItemText>Forum</BreadcrumbsItemText>
-      ) : (
-        <BreadcrumbsItemLink to={ForumRoutes.forum}>Forum</BreadcrumbsItemLink>
-      )}
-
-      {breadcrumbs.map(({ id, title }) => (
+  const allItems = useMemo(
+    () =>
+      breadcrumbs.map(({ id, title }) => (
         <BreadcrumbsItem key={id} url={`${ForumRoutes.category}/${id}`} isLink={!nonInteractive}>
           {title}
         </BreadcrumbsItem>
-      ))}
+      )),
+    [isLoading, nonInteractive]
+  )
+
+  const containerRef = useRef<HTMLUListElement>(null)
+  const [isShortened, shorten] = useState(false)
+  const [keptItemOnEachSide, setKeptItemOnEachSide] = useState<number | undefined>()
+
+  const items = useMemo(
+    () =>
+      isShortened && isDefined(keptItemOnEachSide)
+        ? [
+            ...allItems.slice(0, keptItemOnEachSide),
+            ellipsisItem,
+            ...allItems.slice(allItems.length - 1 - keptItemOnEachSide),
+          ]
+        : allItems,
+    [allItems, keptItemOnEachSide, isShortened]
+  )
+
+  useEffect(() => {
+    if (isLoading || breadcrumbs.length <= 1) return
+
+    const shortenBreadcrumbs = debounce(() => {
+      if (!containerRef.current) return
+
+      const containerBox = containerRef.current.getBoundingClientRect()
+      const children = Array.from(containerRef.current.children)
+      const lastBox = last(children).getBoundingClientRect()
+      const shouldShorten = containerBox.right < lastBox.right
+
+      shorten(shouldShorten)
+      if (!shouldShorten) return
+
+      const lastChildWidth = Math.min(lastBox.width, MAX_BREADCRUMB_WIDTH)
+      const firstChildWidth = Math.min(children[0].getBoundingClientRect().width, MAX_BREADCRUMB_WIDTH)
+      const middleChildrenWidth = children
+        .slice(1, -1)
+        .map((element) => Math.min(element.getBoundingClientRect().width, MIN_BREADCRUMB_WIDTH))
+        .reduce((a, b) => a + b)
+      const childrenWidth = firstChildWidth + middleChildrenWidth + lastChildWidth
+
+      const widthDifference = containerBox.width - childrenWidth
+
+      if (widthDifference >= 0) {
+        setKeptItemOnEachSide(undefined)
+      } else {
+        const totalExtraChildren = Math.ceil(-widthDifference / MIN_BREADCRUMB_WIDTH)
+        setKeptItemOnEachSide(Math.floor((children.length - 2 - totalExtraChildren) / 2))
+      }
+    }, 1000)
```

Comment:
Wow, that actually quite complex. I see that this works quite well. However, I feel that this is a bit too much for our needs ATM. Since it is already here, I don't see a reason to remove that.